### DISCUSSION TITLE
DLL Errors

### DISCUSSION THREAD

**Discussion by c70sp** [opentrack/opentrack#1858] [Q&A] [answered]
Hello everybody, 😄 
I was having problems with the neural network input since it was not showing up, I tried a lot to solve the problem myself but it just didn't want to work.
When launching opentrack via a cmd window, I am getting the following errors:

`DEBUG [F:\dev\opentrack\api/plugin-support.hpp:106]: library "kinect-face" failed: "Cannot load library C:\\Users\\Administrator\\Downloads\\install\\modules\\\\opentrack-tracker-kinect-face.dll: Das angegebene Modul wurde nicht gefunden."`
Translation: The mentioned Module was not found (even though the DLL is in the correct dir and has the correct name)
and
`DEBUG [F:\dev\opentrack\api/plugin-support.hpp:106]: library "neuralnet" failed: "Cannot load library C:\\Users\\Administrator\\Downloads\\install\\modules\\\\opentrack-tracker-neuralnet.dll: Eine DLL-Initialisierungsroutine ist fehlgeschlagen."`
Translation: A DLL initialization routine failed

After replacing onnxruntime.dll with the latest 32bit version, I am only getting the first error again and I can see neuralnet option in the input dropdown. But when starting the tracking with it, the program crashes pretty much instantly.

Any help is appreciated!

**Comment by sthalik** [comment, marked as answer]
Which Windows version are you using? Windows 7 or 10?

Try replacing onnxruntime's DLL with 1.16.1. The next opentrack version will use 1.18.0 though.

**Reply by sthalik** [reply]
Also try using this build: https://ananke.misaki.pl/opentrack/opentrack-test-20240212_01.7z

**Reply by c70sp** [reply]
Currently using windows 10, 64 bit.
Replaced onnxruntime's DLL with 1.16.1 in the opentrack version I had installed and the one you've provided, getting the following error for both
`DEBUG [F:\dev\opentrack\api/plugin-support.hpp:106]: library "neuralnet" failed: "Cannot load library C:\\Program Files (x86)\\opentrack\\modules\\\\opentrack-tracker-neuralnet.dll: Unknown error 0x000000c1."`
The instal you've provided has the same error as in the original post when not replacing the DLL.

Edit: Accidentally used the x64 version when trying just now. Using the 1.16.1 x86 works with both builds.
(Though I still get the error that kinect-face is not found, which is not a problem since don't use it)

Thank you very much!
